experimental/Schemes/BlowupMorphism.jl

```diff
@@ -0,0 +1,255 @@
+export BlowupMorphism
+export center, projection, exceptional_divisor
+
+mutable struct BlowupMorphism{
+                              CodomainType<:AbsCoveredScheme
+                             }
+  projective_bundle::CoveredProjectiveScheme 
+  codomain::CodomainType   # i.A. ein CoveredScheme
+  center::IdealSheaf      # on codomain
+  projection::AbsCoveredSchemeMorphism
+  domain::AbsCoveredScheme # i.A. ein CoveredScheme
+  exceptional_divisor::EffectiveCartierDivisor
+
+  function BlowupMorphism(
+      IP::CoveredProjectiveScheme,
+      I::IdealSheaf
+    )
+    X = base_scheme(IP)
+    X === scheme(I) || error("ideal sheaf not compatible with blown up variety")
+    return new{typeof(X)}(IP, X, I)
+  end
+end
+
+function domain(p::BlowupMorphism)
+  if !isdefined(p, :domain)
+    p.domain = covered_scheme(p.projective_bundle)
+  end
+  return p.domain
+end
+
+codomain(p::BlowupMorphism) = p.codomain
+center(p::BlowupMorphism) = p.center
+
+function projection(p::BlowupMorphism)
+  if !isdefined(p, :projection)
+    p.projection = covered_projection_to_base(p.projective_bundle)
+  end
+  return p.projection
+end
+
+# TODO: Find better name!
+covered_projective_scheme(p::BlowupMorphism) = p.projective_bundle
+
+@Markdown.doc """
+    exceptional_divisor(p::BlowupMorphism)
+
+For a `BlowupMorphism` ``p : Y → X`` coming from the blowup of an 
+`IdealSheaf` ``ℐ`` on X, return the `EffectiveCartierDivisor` ``E`` 
+on ``Y`` associated to the (relative) tautological bundle ``𝒪(1)``. 
+
+On a pair of charts ``V → U`` of the `covered_scheme` of the 
+`projection` of ``p`` this returns the pullback of the `i`-th 
+generator of ``ℐ(U)`` when ``V`` is the `i-1`-st canonical chart 
+of the local blowup over ``U``.
+"""
+function exceptional_divisor(p::BlowupMorphism)
+  if !isdefined(p, :exceptional_divisor)
+    error("exceptional divisor needs to be cached during construction")
```

Comment:
I thought so too, before I found other instances of code made inactive by a error("not implemented"). It rather looks like missing cleanup to me.

Reply:
No, no typo. As it says, the computation of the divisor has been shifted to another place. I kept the other code behind the error message for the moment in case I still wanted something to draw from. But now this is just one of the sanity checks that you proposed the other day.